Firestore/core/test/unit/api/aggregate_query_test.cc

```diff
@@ -0,0 +1,160 @@
+    return aggregateQuery.aggregates_;
+  }
+};
+
+namespace {
+
+// Assert that the Get method calls GetAggregate method
+// and that the result from GetAggregate is processed
+// appropriately
+TEST(AggregateQuery, GetCallsGetAggregateOk) {
+  // Params for Get: Query and AggregateField vector
+  Query query;
+  AggregateField countAggregateField(AggregateField::OpKind::Count,
+                                     AggregateAlias("count"));
+  std::vector<AggregateField> aggregates{countAggregateField};
+
+  // Mock aggregate field result
+  google_firestore_v1_AggregationResult_AggregateFieldsEntry
+      mockAggregatesField[1];
+  mockAggregatesField[0].key =
+      nanopb::ByteString(absl::string_view("count")).release();
+  mockAggregatesField[0].value.which_value_type =
+      google_firestore_v1_Value_integer_value_tag;
+  mockAggregatesField[0].value.integer_value = 10;
+
+  // Mock ObjectValue result
+  ObjectValue mockObjectValue =
+      ObjectValue::FromAggregateFieldsEntry(mockAggregatesField, 1);
+
+  // Set mocked result on our MockAggregateQuery object
+  MockAggregateQuery aggregateQuery(query, std::move(aggregates),
+                                    StatusOr<ObjectValue>(mockObjectValue));
+
+  // Call the Get method
+  size_t callbackCount = 0;
+  aggregateQuery.Get([&callbackCount](const StatusOr<int64_t>& result) mutable {
+    callbackCount++;
+    EXPECT_EQ(result.ok(), true);
+    EXPECT_EQ(result.ValueOrDie(), 10);
+  });
+
+  // Assert
+  EXPECT_EQ(callbackCount, 1);
+}
+
+// Assert that the Get method calls GetAggregate method
+// and that an error result from GetAggregate is processed
+// appropriately
+TEST(AggregateQuery, GetCallsGetAggregateError) {
+  // Params for Get: Query and AggregateField vector
+  Query query;
+  AggregateField countAggregateField(AggregateField::OpKind::Count,
+                                     AggregateAlias("count"));
+  std::vector<AggregateField> aggregates{countAggregateField};
+
+  // Mock error status
+  Status mockError = Status(Error::kErrorInternal, "foo");
+
+  // Set mocked result on our MockAggregateQuery object
+  MockAggregateQuery aggregateQuery(query, std::move(aggregates),
+                                    StatusOr<ObjectValue>(mockError));
+
+  // Call the Get method
+  size_t callbackCount = 0;
+  aggregateQuery.Get([&callbackCount](const StatusOr<int64_t>& result) mutable {
+    callbackCount++;
+    EXPECT_EQ(result.ok(), false);
+    EXPECT_EQ(result.status().code(), Error::kErrorInternal);
+    EXPECT_EQ(result.status().error_message(), "foo");
+  });
+
+  // Assert
+  EXPECT_EQ(callbackCount, 1);
+}
+
+// Assert that the Query.Count method creates an AggregateQuery
+// with the expected query and aggregates
+TEST(Query, Count) {
+  // Baseline Query
+  Query query;
+
+  // Testing the Count() method
+  AggregateQuery aggregateQuery = query.Count();
+
+  auto internalQuery = AggregateQueryTest::GetQuery(aggregateQuery);
```

Comment:
Please don't use `auto` here because it detracts from readability. I have no idea what these types are without somehow finding the declaration. I typically recommend to only use `auto` when the type is immediately obvious from the context, such as the return value of `std::make_shared()`.